src/com/okta/ConsoleClient.java

```diff
@@ -177,17 +218,29 @@ public static void main(String[] args) {
             }
 
             AttributeFactory.loadAttributeDictionary("net.jradius.dictionary.AttributeDictionaryImpl");
+            AttributeFactory.loadAttributeDictionary(new CustomAttributeDictionary());
 
             Pair<String, Integer> hp = hostAndPort(argMap.get(Params.server));
             InetAddress host = InetAddress.getByName(hp.k);
-            RadiusClient rc = new RadiusClient(host, argMap.get(Params.secret), hp.v, 1813, 10);
+            RadiusClient rc = new RadiusClient(host, argMap.get(Params.secret), hp.v, radiusAccountServerPort, timeOut);
 
             RadiusRequest request = new AccessRequest(rc, getBaseAttrList(argMap));
             request.addAttribute(new Attr_UserPassword(argMap.get(Params.password)));
 
             System.out.println("Sending:\n" + request.toString());
 
-            RadiusResponse reply = rc.authenticate((AccessRequest) request, new SmsAuthenticator(rc), 5);
+            RadiusAuthenticator authenticator = null;
```

Comment:
Though SmsAuthenticator was a poor name - it was handling the factor selection and all other factors, not just SMS, correctly. So I am wondering what is the need for DefaultAuthenticator, MfaSelectorAuthenticator and QuestionAuthenticator.